pkg/apis/core/validation/validation.go

```diff
@@ -3600,6 +3600,13 @@ func validateHostUsers(spec *core.PodSpec, fldPath *field.Path) field.ErrorList
 		return allErrs
 	}
 
+	// Fail validation if the kubelet doesn't have the feature on.
+	// This branches from the behavior of other fields because it tightens the security of a pod,
+	// and the kube-apiserver may relax validation for a pod assuming it is being confined by a user namespace when it is not.
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
```

Comment:
/hold

This will break updates of data that was previously persisted when the gate was enabled.

In general, checking feature gates in validation to control accepted values is a sign we are breaking validation.

The current approach (dropping the unknown field) matches what an old server would do when presented with an unknown field and is described in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version